zetaclient/bitcoin_client.go

```diff
@@ -1024,13 +1020,24 @@ func (ob *BitcoinChainClient) observeOutTx() {
 				if len(tracker.HashList) > 1 {
 					ob.logger.ObserveOutTx.Warn().Msgf("observeOutTx: oops, outTxID %s got multiple (%d) outTx hashes", outTxID, len(tracker.HashList))
 				}
-				// verify outTx hashes
+				// iterate over all txHashes to find the truly included one.
+				// we do it this (inefficient) way because we don't rely on the first one as it may be a false positive (for unknown reason).
+				txCount := 0
+				var txResult *btcjson.GetTransactionResult
 				for _, txHash := range tracker.HashList {
-					_, err := ob.checkNSaveIncludedTx(txHash.TxHash, params)
-					if err != nil {
-						ob.logger.ObserveOutTx.Error().Err(err).Msg("observeOutTx: checkNSaveIncludedTx failed")
+					result, inMempool := ob.checkIncludedTx(txHash.TxHash, params)
+					if result != nil && !inMempool { // included
+						txCount++
+						txResult = result
+						ob.logger.ObserveOutTx.Info().Msgf("observeOutTx: included outTx %s for chain %d nonce %d", txHash.TxHash, ob.chain.ChainId, tracker.Nonce)
 					}
 				}
+				if txCount == 1 { // should be only one txHash included for each nonce
+					ob.setIncludedTx(tracker.Nonce, txResult)
+				} else if txCount > 1 {
+					ob.removeIncludedTx(tracker.Nonce) // we can't tell which txHash is true, so we remove all (if any) to be safe
+					ob.logger.ObserveOutTx.Error().Msgf("observeOutTx: confirmed multiple (%d) outTx for chain %d nonce %d", txCount, ob.chain.ChainId, tracker.Nonce)
```

Comment:
Maybe we can also log all results?

Reply:
[commit](https://github.com/zeta-chain/node/pull/1530/commits/4d24ba320d56bdbddb1bb666246495503df896af)